src/object.c

```diff
@@ -208,9 +209,58 @@ int write_object_file(SDL_RWops *rw, rct_object_entry* entry)
 	chunkHeader.encoding = object_entry_group_encoding[type];
 	chunkHeader.length = installed_entry->chunk_size;
 
-	size_dst += sawyercoding_write_chunk_buffer(dst_buffer + sizeof(rct_object_entry), chunk, chunkHeader);
-	SDL_RWwrite(rw, dst_buffer, 1, size_dst);
 
+	//Check if content of object file matches the stored checksum. If it does not, then fix it.
+	int calculated_checksum=object_calculate_checksum(entry,chunk,installed_entry->chunk_size);
+		if(entry->checksum!=calculated_checksum)
```

Comment:
Indentation suggests how the code flows, so a child block is more indented than the parent.

```
// parent scope
int x = 3;
if (x == 2) {
    // indented child scope
    x++;
} else {
    x--;
}
```